angular/projects/researchdatabox/portal-ng-common/src/lib/handlebars-template.service.ts

```diff
@@ -0,0 +1,322 @@
+            }
+        });
+
+        // Parse date string helper (for chaining)
+        Handlebars.registerHelper('parseDateString', (dateString: string) => {
+            return dateString;
+        });
+
+        // Get nested property helper
+        Handlebars.registerHelper('get', (obj: any, path: string) => {
+            return _get(obj, path, '');
+        });
+
+        // Equality comparison helper
+        Handlebars.registerHelper('eq', (a: any, b: any) => {
+            return a === b;
+        });
+
+        // Not equal helper
+        Handlebars.registerHelper('ne', (a: any, b: any) => {
+            return a !== b;
+        });
+
+        // And helper
+        Handlebars.registerHelper('and', (...args: any[]) => {
+            args.pop(); // Remove options hash
+            return args.every(Boolean);
+        });
+
+        // Or helper 
+        Handlebars.registerHelper('or', (...args: any[]) => {
+            args.pop(); // Remove options hash
+            return args.some(Boolean);
+        });
+
+        // isEmpty helper
+        Handlebars.registerHelper('isEmpty', (value: any) => {
+            return _isEmpty(value);
+        });
+
+        // isUndefined helper
+        Handlebars.registerHelper('isUndefined', (value: any) => {
+            return _isUndefined(value);
+        });
+
+        // isNull helper
+        Handlebars.registerHelper('isNull', (value: any) => {
+            return _isNull(value);
+        });
+
+        // isDefined helper (not undefined and not null)
+        Handlebars.registerHelper('isDefined', (value: any) => {
+            return !_isUndefined(value) && !_isNull(value);
+        });
+
+        // Translation helper - will be resolved via context.translationService
+        Handlebars.registerHelper('t', function (this: any, key: string) {
+            const translationService = this.translationService;
+            if (translationService && typeof translationService.t === 'function') {
+                return translationService.t(key);
+            }
+            return key;
+        });
+
+        this.helpersRegistered = true;
+        this.loggerService.debug('Handlebars helpers registered');
+    }
+
+    /**
+     * Load pre-compiled templates from the server for a specific record type and workflow stage.
+     * Templates are loaded as ES modules using dynamic import.
+     * 
+     * @param branding The branding name
+     * @param portal The portal name  
+     * @param recordType The record type name
+     * @param workflowStage The workflow stage name
+     * @param dashboardType Optional dashboard type to load specific configuration
+     */
+    public async loadDashboardTemplates(branding: string, portal: string, recordType: string, workflowStage: string, dashboardType: string = 'standard'): Promise<void> {
+        const cacheKey = `${branding}/${portal}/${recordType}/${workflowStage}/${dashboardType}`;
+
+        if (this.loadedConfigs.has(cacheKey)) {
```

Comment:
I think the caching makes more sense in the `utilService.getDynamicImport` method, but it's not so important right now.